azurerm/internal/services/postgres/postgresql_flexible_server_configuration.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/postgres/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/postgres/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourcePostgresqlFlexibleServerConfiguration() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceFlexibleServerConfigurationUpdate,
+		Read:   resourceFlexibleServerConfigurationRead,
+		Update: resourceFlexibleServerConfigurationUpdate,
+		Delete: resourceFlexibleServerConfigurationDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FlexibleServerConfigurationID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"server_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FlexibleServerName,
+			},
+
+			"value": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceFlexibleServerConfigurationUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Postgres.FlexibleServersConfigurationsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure Postgresql Flexible Server configuration creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	serverName := d.Get("server_name").(string)
+
+	props := postgresqlflexibleservers.Configuration{
+		ConfigurationProperties: &postgresqlflexibleservers.ConfigurationProperties{
+			Value:  utils.String(d.Get("value").(string)),
+			Source: utils.String("user-override"),
+		},
+	}
+
+	future, err := client.Update(ctx, resGroup, serverName, name, props)
+	if err != nil {
+		return err
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, resGroup, serverName, name)
+	if err != nil {
+		return err
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Azure Postgresql Flexible Server configuration %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*resp.ID)
```

Comment:
We should rather construct a `FlexibleServerConfigurationId` and do `d.SetId(id.ID())`, then we can remove the extra Get operation after the create/update.

Reply:
now we construct FlexibleServerConfigurationId and use it in d.SetId(id.ID())